### PR TITLE
Wallet: activity items

### LINKED ISSUE #19825: Wallet: Activity Tab - fetch and display activity items
This is the first issue for the activity tab epic.

The scope of this issue to to fetch from status-go the activity items and display the list of activity items for an account.

Designs: https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=1070-283956&mode=design&t=BFnqkraLW9DQD1GG-4

### PR DESCRIPTION
**Pull request by OmarBasem**
fixes: https://github.com/status-im/status-mobile/issues/19825

epic: https://github.com/status-im/status-mobile/issues/19849

This is the first PR for the activity items Epic. This PR implements fetching and displaying the list of activity items under the activity tab inside a wallet account. I have also removed the activity tab from the wallet home.

[Designs.](https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=1070-284225&mode=design&t=05G37UXdKebXGVOI-4)

Demo:

https://github.com/status-im/status-mobile/assets/29354102/d56c8300-0501-4248-93b8-f4a3e1e40e50

Thanks to @smohamedjavid for doing some of the initial setup of the activity events.

### CODE REVIEW

File: src/status_im/contexts/wallet/common/activity_tab/view.cljs [hunk 1]
```diff
@@ -1,24 +1,68 @@
...
-    (merge {:on-press #(js/alert "Item pressed")}
-           item)]])
+  [{:keys [activity-type activity-status timestamp symbol-out symbol-in token-in token-out amount-in
+           amount-out sender recipient]}]
+  (let [chain-id         (or (:chain-id token-in) (:chain-id token-out))
+        amount-in-units  (native-module/hex-to-number
+                          (utils.hex/normalize-hex amount-in))
+        amount-in-value  (money/with-precision
+                          (money/wei->ether amount-in-units)
+                          6)
```
Comment (vkjr) [inline src/status_im/contexts/wallet/common/activity_tab/view.cljs]: We can extract magic number to def it with meaningful name

File: src/status_im/contexts/wallet/common/activity_tab/view.cljs [hunk 2]
```diff
@@ -1,24 +1,68 @@
...
+        relative-date    (datetime/timestamp->relative (* timestamp 1000))]
+    [quo/wallet-activity
+     {:transaction       (constants/wallet-activity-id->name activity-type)
+      :timestamp         relative-date
+      :status            (constants/wallet-activity-status->name activity-status)
+      :counter           1
+      :first-tag         {:size   24
+                          :type   :token
+                          :token  (or symbol-out symbol-in)
+                          :amount (if (= activity-type constants/wallet-activity-type-receive)
```
Comment (vkjr) [inline src/status_im/contexts/wallet/common/activity_tab/view.cljs]: Result of comparison `(= activity-type constants/wallet-activity-type-receive)` can be assigned within `let` and reused 3 times

File: src/status_im/contexts/wallet/events.cljs [hunk 3]
```diff
@@ -479,3 +496,45 @@
...
+         offset         0
+         limit          20
+         request-params [request-id
+                         addresses
+                         chain-ids
+                         filters
+                         offset
+                         limit]]
+     {:fx [[:json-rpc/call
+            [{;; This method is deprecated and will be replaced by
```
Comment (vkjr) [inline src/status_im/contexts/wallet/events.cljs:526]: Maybe we can create an issue for this and link it here, wdyt?

**Review by vkjr — APPROVED** [review]

File: src/quo/components/wallet/wallet_activity/view.cljs [hunk 4]
```diff
@@ -118,10 +118,10 @@
       :on-press-out        on-press-out}
      [rn/view
       {:style {:flex-direction :row}}
-      [transaction-icon-view props]
+      [transaction-icon-view props theme]
       [rn/view
        {:style style/content-container}
-       [transaction-header props]
+       [transaction-header props theme]
```
Comment (shivekkhurana) [inline src/quo/components/wallet/wallet_activity/view.cljs:124]: Is there any reasoning for moving themes outside of props map?

File: src/status_im/contexts/wallet/account/view.cljs [hunk 5]
```diff
@@ -53,7 +53,10 @@
            :size             32
            :default-active   @selected-tab
            :data             (tabs-data watch-only?)
-           :on-change        #(reset! selected-tab %)
+           :on-change        (fn [tab]
```
Comment (shivekkhurana) [inline src/status_im/contexts/wallet/account/view.cljs]: Are you sure this needs no memoization?

**Review by shivekkhurana — CHANGES_REQUESTED** [review]
Thank you for the PR. Looks good overall, left some comments.

image

File: src/quo/components/wallet/wallet_activity/view.cljs (hunk 4 above)
Comment (OmarBasem) [reply src/quo/components/wallet/wallet_activity/view.cljs:124]: @shivekkhurana we no longer use the old theme approach where it was in the props. We now get theme from 
`theme         (quo.theme/use-theme)`

Refactored by @flexsurfer in that PR https://github.com/status-im/status-mobile/pull/19706 and this is a leftover
Example current usage: https://github.com/status-im/status-mobile/blob/develop/src/quo/components/markdown/text.cljs#L9

File: src/status_im/contexts/wallet/account/view.cljs (hunk 5 above)
Comment (OmarBasem) [reply src/status_im/contexts/wallet/account/view.cljs]: @shivekkhurana there was no memoization. Do you think it needs memoization?

Comment (shivekkhurana) [reply src/status_im/contexts/wallet/account/view.cljs]: I'm not sure, you need to check in the children that are getting rendered.

Comment (OmarBasem) [reply src/status_im/contexts/wallet/account/view.cljs]: @shivekkhurana fixed it! 👍

**Review by shivekkhurana — APPROVED** [review]

**Pull request merged**